Src/SemanticComparison/ProxyGenerator.cs

```diff
@@ -268,13 +269,10 @@ private static void BuildMethodEquals(TypeBuilder type, FieldInfo comparer)
             var methodAttributes = MethodAttributes.Public | MethodAttributes.Virtual | MethodAttributes.HideBySig;
             MethodBuilder method = type.DefineMethod("Equals", methodAttributes);
 
-            MethodInfo equals = typeof(object).GetMethod(

```

Comment:
Notice a change here - now only public method will be matched. That should be fine as method is public.